interface-impl-amf/src/main/java/org/mule/amf/impl/model/ParameterImpl.java

```diff
@@ -47,6 +48,14 @@ class ParameterImpl implements Parameter {
   private Set<String> allowedEncoding;
   private boolean required;
 
+  private LazyValue<Boolean> isArray = new LazyValue<>(() -> {
+    if (schema instanceof UnionShape) {
+      UnionShape unionShape = (UnionShape) schema;
+      return unionShape.anyOf().stream().allMatch(shape -> shape instanceof ArrayShape);
```

Comment:
Should we case about something like `string[] | SomeOtherType` where `SomeOtherType` is defined as `number[] | boolean[]`?

Reply:
I think this doesn't hurt s

Reply:
Actually that could be cyclic and handle that sounds like an overkill, but I introduce a fix for nullable arrays.

Reply:
You are right, implementing the correct behaviour for `type Thing = number[] | Thing` sounds a bit tricky.
You can always keep a set of visited types and try not to visit the same type twice (like when doing a DFS or BFS graph traversal) but it would turn illegible quite quickly.